src/platform/openiotsdk/Logging.cpp

```diff
@@ -59,23 +59,21 @@ char logMsgBuffer[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
  */
 void ENFORCE_FORMAT(3, 0) LogV(const char * module, uint8_t category, const char * msg, va_list v)
 {
-    size_t prefixLen = 0;
-    snprintf(logMsgBuffer, sizeof(logMsgBuffer), "[%s]", module);
+    size_t msglen = 0;
+    snprintf(logMsgBuffer, sizeof(logMsgBuffer), "[%s] ", module);
     logMsgBuffer[sizeof(logMsgBuffer) - 2] = 0; // -2 to allow at least one char for the vsnprintf
-    prefixLen                              = strlen(logMsgBuffer);
-    vsnprintf(logMsgBuffer + prefixLen, sizeof(logMsgBuffer) - prefixLen, msg, v);
+    msglen                                 = strlen(logMsgBuffer);
+    vsnprintf(logMsgBuffer + msglen, sizeof(logMsgBuffer) - msglen, msg, v);
+    logMsgBuffer[sizeof(logMsgBuffer) - 2] = 0;
+    msglen                                 = strlen(logMsgBuffer);
+    strncat(logMsgBuffer, "\r\n", sizeof(logMsgBuffer) - msglen);
```

Comment:
I mean something like this:
```suggestion
    char * endline = "\r\n";
    if (strlen(endline) + strlen(module) + 2 > sizeof(logMsgBuffer))
    {
        return;
    }
    size_t msglen = strlen(endline);
    snprintf(logMsgBuffer, sizeof(logMsgBuffer) - msglen, "[%s] ", module);
    msglen                                 = strlen(module);
    vsnprintf(logMsgBuffer + msglen, sizeof(logMsgBuffer) - msglen - strlen(endline), msg, v);
    msglen                                 = strlen(logMsgBuffer);
    strncat(logMsgBuffer, endline, sizeof(logMsgBuffer) - msglen);
```

Reply:
Ok now the function returns if the string is too long to be fully printed